tests/pthread/test_pthread_proxying_refcount.c

```diff
@@ -0,0 +1,69 @@
+#include <assert.h>
+#include <emscripten/console.h>
+#include <emscripten/emscripten.h>
+#include <pthread.h>
+#include <unistd.h>
+
+#include "proxying.h"
+
+em_proxying_queue* queue;
+
+_Atomic int should_execute = 0;
+_Atomic int executed = 0;
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
+
+void task(void* arg) {
+  executed = 1;
+  pthread_cond_signal(&cond);
+}
+
+void* execute_and_free_queue(void* arg) {
+  // Wait until we are signaled to execute the queue.
+  pthread_mutex_lock(&mutex);
+  while (!should_execute) {
+    pthread_cond_wait(&cond, &mutex);
+  }
+  pthread_mutex_unlock(&mutex);
+
+  // Execute the proxied work then free the empty queue.
+  emscripten_proxy_execute_queue(queue);
+  em_proxying_queue_destroy(queue);
+
+  // Exit with a live runtime so the queued work notification is received and we
+  // try to execute the queue again, even though it has been destroyed.
+  emscripten_exit_with_live_runtime();
+}
+
+int main() {
+  emscripten_console_log("start");
+  queue = em_proxying_queue_create();
+  assert(queue);
+
+  // Create the worker and send it a task.
+  pthread_t worker;
+  pthread_create(&worker, NULL, execute_and_free_queue, NULL);
+  emscripten_proxy_async(queue, worker, task, NULL);
+  should_execute = 1;
+  pthread_cond_signal(&cond);
+
+  // Wait for the task to be executed.
+  pthread_mutex_lock(&mutex);
+  while (!executed) {
+    pthread_cond_wait(&cond, &mutex);
+  }
+  pthread_mutex_unlock(&mutex);
```

Comment:
I was wondering if would make more sense just to busy wait on the atomic?  I just converted a test to that style.. because I wanted to explicitly avoid futex_wait.. but I think it makes the test more readable too: #16525

Reply:
Yeah, I can do that. I'll update this test to wrap malloc to assert that the queues are properly freed as well.

Reply:
I don't feel strongly about it